packages/pirania/Readme.md

```diff
@@ -99,13 +119,81 @@ It uses iptables rules to filter inbound connections outside the mesh network.
 ```
 files/
     /etc/config/pirania is the UCI config
-    /etc/pirania/db.csv (default path) contains the database of vouchers
+    /etc/pirania/vouchers/ (default path) contains the database of vouchers
     /etc/init.d/pirania-uhttpd starts a uhttpd on port 59080 that replies any request with a redirect towards a preset URL
 
     /usr/lib/lua/voucher/ contains lua libraries used by /usr/bin/voucher
-    /usr/bin/voucher is a CLI to manage the db (has functions add_voucher, add_many_vouchers, auth_voucher, get_valid_macs, list_vouchers, remove_voucher and url)
+    /usr/bin/voucher is a CLI to manage the db (has functions show, show_active, show_authorized_macs, add, activate, deactivate and is_mac_authorized)
     /usr/bin/captive-portal sets up iptables rules to capture traffic
 
     /usr/libexec/rpcd/pirania ubus pirania API (this is used by the web frontend)
     /usr/share/rpcd/acl.d/pirania.json ACL for the ubus pirania API
-```
\ No newline at end of file
+
+    /etc/shared-state/publishers/shared-state-publish_vouchers inserts into shared-state the local voucher db
+    /etc/shared-state/hooks/pirania/generate_vouchers bring updated or new vouchers from the shared-state database into the local voucher db
+```
+
+### Trafic capture
+`/usr/bin/captive-portal` sets up iptables rules to capture traffic.
+It creates a set of rules that apply to 3 allowed "ipsets":
+* pirania-auth-macs: authorized macs go into this rule. starts empty.
+* pirania-allowlist-ipv4: with the members of the allowlist in the config file (10.0.0.0/8, 192.168.0.0/16, 172.16.0.0/12)
+* pirania-allowlist-ipv6: same as ipv4 but for ipv6
+
+Rules:
+* DNS packets, that are not from the allowed ipsets, are redirected to our own captive portal DNS at 59053
+* HTTP packets, that are not from the allowed ipsets, are redirected to our own captive portal HTTP at 59080
+* packets from the allowed ipsets are allowed
+* the rest of the packets are rejeted (drop and send an error to the client)
+
+### HTTP flow
```

Comment:
Yes, it should not be in a Readme but at the same time we are using the readme of the packages as documentation. Maybe when we implement an automatic generation of the full docs of libremesh it may be good idea to remove this details from the Readme but for the time being I think that nobody will see it (and I will forget about it).